doc/source/coding-style/pep8.rst

```diff
@@ -0,0 +1,813 @@
+
+.. tabs::
+
+    .. tab:: Avoid
+    
+        .. code-block:: python
+        
+            from my_package.my_module import *
+    
+    .. tab:: Use
+    
+        .. code-block:: python
+        
+            from my_package.my_module import myclass
+
+
+Naming Conventions
+------------------
+To achieve readable and maintainable code, use concise and descriptive names for classes,
+methods, functions, and constants. Regardless of the programming language, you must follow these
+global rules to determine the correct names:
+
+#. Choose descriptive and unambiguous names.
+#. Make meaningful distinctions.
+#. Use pronounceable names.
+#. Use searchable names.
+#. Replace magic numbers with named constants.
+#. Avoid encodings. Do not append prefixes or type information.
+
+
+Variables
+~~~~~~~~~
+Do not use the characters ``'l'``, ``'O'`` , or ``'I'`` as single-character
+variable names. In some fonts, these characters are indistinguishable from the
+numerals one and zero.
+
+
+Packages and Modules
+~~~~~~~~~~~~~~~~~~~~
+Use a short, lowercase word or words for module names. Separate words
+with underscores to improve readability. For example, use ``module.py``
+or ``my_module.py``.
+
+For a package name, use a short, lowercase word or words. Avoid
+underscores as these must be represented as dashes when installing
+from PyPi.
+
+.. code::
+
+   python -m pip install package
+
+
+Classes
+~~~~~~~
+Use `camel case <https://en.wikipedia.org/wiki/Camel_case>`_ when naming
+classes. Do not separate words with underscores. 
+
+.. code:: python
+
+   class MyClass:
+       """Docstring for MyClass"""
+
+       ...
+
+
+Use a lowercase word or words for Python functions or methods. Separate words
+with underscores to improve readability. When naming class methods, the
+following conventions apply:
+
+- Enclose only `dunder methods`_ with double underscores.
+- Start a method that is to be considered private with double underscores.
+- Start a method that is to be considered protected with a single underscore.
+
+.. _dunder methods: https://docs.python.org/3/reference/datamodel.html#special-method-names
+
+.. code:: python
+
+   class MyClass:
+       """Docstring for MyClass."""
+
+       def __init__(self, value):
+           """Constructor.
+
+           Methods with double underscores on either side are called
+           "dunder" methods and are special Python methods.
+
+           """
+           self._value = value
+
+       def __private_method(self):
+           """This method can only be called from ``MyClass``."""
+           self._value = 0
+
+       def _protected_method(self):
+           """This method should only be called from ``MyClass``.
+
+           Protected methods can be called from inherited classes,
+           unlike private methods, which names are 'mangled' to avoid
+           these methods from being called from inherited classes.
```

Comment:
```suggestion
           Protected methods can be called from inherited classes.
           For private methods, names are 'mangled' to prevent
           these methods from being called from inherited classes.
```